modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/WorkflowInstance.java

```diff
@@ -82,6 +82,32 @@
                 name = "Workflow.findAll",
                 query = "select w from WorkflowInstance w where w.organizationId=:organizationId order by w.dateCreated"
         ),
+        // Finding the latest or active workflow for all events
+        @NamedQuery(
+                name = "Workflow.findLatest",
+                query = "SELECT w FROM WorkflowInstance w "
+                      + "WHERE "
+                      + "w.organizationId=:organizationId AND "
+                      + "((w.state = :stateInstantiated OR w.state = :statePaused OR w.state = :stateRunning OR w.state = :stateFailing) "
+                      + "OR "
+                      + "w.dateCompleted = (SELECT MAX(w2.dateCompleted) "
+                      + "FROM WorkflowInstance w2 "
+                      + "WHERE w2.mediaPackageId = w.mediaPackageId) "
+                      + ") "
+                      + "GROUP BY w.mediaPackageId"
```

Comment:
I don't think this `grooup by` statement works with PostgreSQL. Did you resr rgar?
But then, I don't think it's actually necessary at all.